Mathlib/Analysis/Normed/Module/Complemented.lean

```diff
@@ -128,4 +128,119 @@ theorem ClosedComplemented.of_quotient_finiteDimensional [CompleteSpace 𝕜]
   haveI : FiniteDimensional 𝕜 q := (p.quotientEquivOfIsCompl q hq).finiteDimensional
   exact .of_isCompl_isClosed hq hp q.closed_of_finiteDimensional
 
+variable (p q)
+
+/-- Idempotent corresponding to a complemented subspace. -/
+def idempotentOfClosedCompl (h : IsCompl p q) (hp : IsClosed (p : Set E))
+    (hq : IsClosed (q : Set E)) : E →L[𝕜] E :=
+  ↑(prodEquivOfClosedCompl p q h hp hq) ∘L (ContinuousLinearMap.inl 𝕜 p q ∘L
+    (linearProjOfClosedCompl p q h hp hq))
+
+variable {p q}
+
+lemma mem_iff_invariant_ofClosedCompl  (h : IsCompl p q) (hp : IsClosed (p : Set E))
+    (hq : IsClosed (q : Set E)) {x : E} :
+    let P := idempotentOfClosedCompl p q h hp hq
```

Comment:
here and throughout, you want to inline `P` since the only purpose is to make the statement more readable (although I'm not sure it's worth it in this PR since `P` only appears once in each statement).
```suggestion
    letI P := idempotentOfClosedCompl p q h hp hq
```

Reply:
I found it made the statements more readable - but on reflection I should use comments for that.